source/dub/commandline.d

```diff
@@ -630,20 +631,25 @@ class CommandArgs {
 
 	void getopt(T)(string names, T* var, string[] help_text = null, bool hidden=false)
 	{
+		import std.traits : OriginalType;
+
 		foreach (ref arg; m_recognizedArgs)
 			if (names == arg.names) {
 				assert(help_text is null, format!("Duplicated argument '%s' must not change helptext, consider to remove the duplication")(names));
-				*var = arg.value.get!T;
+				*var = arg.value.match!(
+					(OriginalType!T v) => cast(T)v,
+					(_) => assert(false, "value from previous getopt has different type than the current getopt call"))
+				;
 				return;
 			}
 		assert(help_text.length > 0);
 		Arg arg;
-		arg.defaultValue = *var;
+		arg.defaultValue = Arg.Value(cast(OriginalType!T)*var);
```

Comment:
Shouldn't implicit conversion take care of this, ditto below ?

Reply:
`cast(OriginalType!T)` is important because SumType does not do implicit type conversions. I can leave out the constructor call though and rely on opAssign.